common_helper_extraction/fs.py

```diff
@@ -0,0 +1,64 @@
+'''
+    common_helper_extraction
+    Copyright (C) 2018-2019  Fraunhofer FKIE
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+import logging
+from struct import calcsize, unpack
+
+SQFS_MAGIC_STRINGS = [b'sqsh', b'qshs', b'shsq', b'hsqs']
+SQFS_SIZE_BUFFER_OFFSET = 0x28
+SQFS_SIZE_BUFFER_TYPE = 'Q'
+
+
+def extract_fs(input_data: bytes, magic_strings: list, size_buffer_offset: int, size_buffer_type: str) -> list:
+    fs_sections = list()
+    for fs_magic in magic_strings:
+        fs_sections.extend(_get_fs_sections_with_magic(input_data, fs_magic, size_buffer_offset, size_buffer_type))
+    return fs_sections
```

Comment:
this could be replaced with a list comprehension (maybe this improves readability):
```python
fs_sections = [
    fs_section
    for fs_magic in magic_strings
    for fs_section in _get_fs_sections_with_magic(input_data, fs_magic, size_buffer_offset, size_buffer_type)
]
```

Reply:
It may be more pyhthonic but I do not think that this improves readability. You replace one for-loop by a hard to understand two for loop expression.